pulser-core/pulser/json/abstract_repr/validation.py

```diff
@@ -13,13 +13,23 @@
 # limitations under the License.
 """Function for validation of JSON serialization to abstract representation."""
 import json
+from importlib.metadata import version
 from typing import Literal
 
 import jsonschema
 from referencing import Registry, Resource
 
-from pulser.json.abstract_repr import SCHEMAS
+from pulser.json.abstract_repr import SCHEMAS, SCHEMAS_PATH
 
+LEGACY_JSONSCHEMA = version("jsonschema") == "4.17.3"
```

Comment:
This should work too:
```suggestion
LEGACY_JSONSCHEMA = ("4.18"  > version("jsonschema") >= "4.17.3")
```
Then we can simply the dependency